distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/common/downloads/DownloadStrategy.java

```diff
@@ -0,0 +1,193 @@
+                }
+            }
+        };
+    }
+
+    /**
+     * Create an asynchronous strategy for the caller and hit the downloads endpoint for status
+     * updates.
+     */
+    static DownloadStrategy Asynchronous() {
+        return new DownloadStrategy() {
+
+            BlockingCacheChecker checker;
+
+            @Override
+            public AbstractIndexBlocker getIndexBlocker() {
+                return checker;
+            }
+
+            @Override
+            public Future<String> startDownload(String metacardId, CyclicBarrier barrier) {
+                CloseableHttpAsyncClient client = HttpAsyncClients.createDefault();
+                client.start();
+
+                Future downloadRequestFuture = client.execute(new HttpGet(getAsyncDownloadUrl(
+                        metacardId)), null);
+                try {
+                    HttpResponse response = (BasicHttpResponse) downloadRequestFuture.get();
+                    int responseCode = response.getStatusLine()
+                            .getStatusCode();
+                    assertThat("Asynchronous request was not successful (responseCode != 200)",
+                            responseCode,
+                            is(200));
+                } catch (InterruptedException e) {
+                    Thread.currentThread()
+                            .interrupt();
+                } catch (ExecutionException e) {
+                    throw new RuntimeException("Exception occurred in future's thread", e);
+                }
+
+                if (downloadRequestFuture.isCancelled()) {
+                    throw new IllegalStateException("Future was cancelled");
+                }
+                if (!downloadRequestFuture.isDone()) {
+                    throw new IllegalStateException("Future has not finished");
+                }
+
+                checker = new BlockingCacheChecker(metacardId, barrier);
+                return checker.begin();
+            }
+
+            @Override
+            public void cancelDownload() {
+                /* TODO: Look into invoking the cancel endpoint */
+                throw new NotImplementedException("Async cancel has not yet been implemented");
+            }
+        };
+    }
+
+    static String getSyncDownloadUrl(String metacardId) {
+        return String.format("%s/catalog/sources/%s/%s?transform=resource",
+                AbstractIntegrationTest.SERVICE_ROOT.getUrl(),
+                CSW_STUB_SOURCE_ID,
+                metacardId);
+    }
+
+    static String getAsyncDownloadUrl(String metacardId) {
+        return String.format("%s?source=%s&metacard=%s",
+                AbstractIntegrationTest.RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl(),
+                CSW_STUB_SOURCE_ID,
+                metacardId);
+    }
+
+    static String getDownloadsListUrl() {
+        return String.format("%s/catalog/downloads", AbstractIntegrationTest.SERVICE_ROOT.getUrl());
```

Comment:
For something simple like this, I don't feel that `String.format` provides any sort of readability improvement. It also hurts your performance for this operation.